src/psyclone/dynamo0p3.py

```diff
@@ -9037,13 +9075,13 @@ def _find_or_create_type(mod_name, type_name):
 
             api_config = Config.get().api_conf("dynamo0.3")
 
-            if self.intrinsic_type == 'real':
-                kind_name = api_config.default_kind["real"]
+            if self.intrinsic_type == "real":
+                kind_name = const.DATA_TYPE_MAP["scalar"]["kind"]
```

Comment:
After investigation by @TeranIvy and myself, it is not possible to do what I suggest due to the way the code is written so I am withdrawing this request.

In summary the problem is that `DynKernelArgument.__init__()` calls `KernelArgument.__init__()` which calls `Argument.__init__()` which sets `self._precision`, `self._data_type` and `self._module_name` to `None`. `Argument.__init__()` then calls `self.infer_datatype()`. Therefore `infer_datatype()` must look up `precision` from the constant/config location, rather than using `self._precision` as that will be `None`. `DynKernelArgument` does set the values of `self._precision` etc but only after `KernelArgument.__init__()` has completed (which is too late for the `infer_datatype()` call. Setting the values before `KernelArgument.__init__()` does not help as they are then set back to `None` in `Argument.__init__()`.

Reply:
Thanks for the summary of investigation. I modified the setting of the scalar properties in `infer_datatype` to differentiate between scalar reductions and read-only scalars as agreed for the initialiser (it had to be done so we don't generate incorrect symbols for the built-in reduction arguments).